backend/core/src/listings/dto/listing.dto.ts

```diff
@@ -848,12 +856,21 @@ export class ListingsRetrieveQueryParams {
   view?: string
 }
 
+const FilterKeysList = { ...ListingFilterKeys, ...AvailabilityFilterEnum }
+type FilterKeysList = typeof FilterKeysList
+
 // Using a record lets us enforce that all types are handled in addFilter
-export const filterTypeToFieldMap: Record<keyof typeof ListingFilterKeys, string> = {
+export const filterTypeToFieldMap: Record<keyof typeof FilterKeysList, string> = {
   status: "listings.status",
   name: "listings.name",
   neighborhood: "property.neighborhood",
   bedrooms: "unitTypeRef.num_bedrooms",
   zipcode: "buildingAddress.zipCode",
   seniorHousing: "reservedCommunityType.name",
+  // Fields for the availability are determined based on the value of the filter, not the
+  // key. Keep this bogus value to prevent the filter from being rejected.
+  availability: "",
```

Comment:
This is a possible filter value? Even though it doesn't change the query at all?

Reply:
Or does this comment mean that this is a valid filter type because we use the keys of this map as validation?

Reply:
`availability` has to be a valid filter, because we send filters like `filter[availability]=hasAvailability`. In the backend, we check the validity of a filter based on its presence in this `filterTypeToFieldMap`, so I put it here. It also lets us keep this type as a Record, to enforce handling all types.

But because the _value_ of the filter determines which column we filter on, we don't need a mapping from `availability` to a column, we only need mappings from the values to their respective columns. So that's why the value is bogus.